DeepSpeech.py

```diff
@@ -307,8 +307,26 @@ def get_tower_results(iterator, optimizer, dropout_rates):
                     tower_avg_losses.append(avg_loss)
 
                     # Compute gradients for model parameters using tower's mini-batch
-                    gradients = optimizer.compute_gradients(avg_loss)
-
+                    if FLAGS.load == "transfer":
+                        if not FLAGS.fine_tune:
+                            # train from source model and freeze old layers
+                            # aka - only update new layers
+                            gradients = optimizer.compute_gradients(
+                                avg_loss,
+                                var_list = [ v for v in tfv1.trainable_variables()
+                                             if any(
+                                                     layer in v.op.name
+                                                     for layer in drop_source_layers
+                                             )]
+                            )
+                        else:
+                            # Transfer learning, but update all layers
+                            # aka - fine tune all layers
+                            gradients = optimizer.compute_gradients(avg_loss)
+                    else:
+                        # no Transfer Learning, and updating all layers 
+                        gradients = optimizer.compute_gradients(avg_loss)
```

Comment:
This block is hard to read. The two `else`s are identical, so why not have a single test for `if FLAGS.load == "transfer" and not FLAGS.fine_tune:`?